tools/dgeni/processors/filter-export-aliases.ts

```diff
@@ -0,0 +1,48 @@
+import {DocCollection, Processor} from 'dgeni';
+import {ExportDoc} from 'dgeni-packages/typescript/api-doc-types/ExportDoc';
+import {ModuleDoc} from 'dgeni-packages/typescript/api-doc-types/ModuleDoc';
+
+/**
+ * Processor to filter out Dgeni documents that are export aliases. Keeping them means
+ * that a given document shows up multiple times in the docs.
+ *
+ * ```ts
+ *   export {ObserveContent} from './X';
+ *   export {ObserveContent as CdkObserveContent} from './X';
+ * ```
+ *
+ * This is a common pattern inside of Angular Material, but causes Dgeni to generate
+ * documents for both exports. The second document is identical to the original document and
+ * doesn't even show the aliased name.
+ *
+ * See: https://github.com/angular/dgeni-packages/issues/248
+ */
+export class FilterExportAliases implements Processor {
+  name = 'filter-export-aliases';
+  $runBefore = ['categorizer'];
+
+  $process(docs: DocCollection) {
+    return docs.filter(doc => this.filterAliasExport(doc));
+  }
+
+  private filterAliasExport(doc: ExportDoc) {
+    if (!doc.moduleDoc) {
+      return true;
+    }
+
+    const moduleDoc = doc.moduleDoc as ModuleDoc;
+    const duplicateDocs = moduleDoc.exports.filter(exportDoc => exportDoc.id === doc.id);
+
+    // Remove the current export document if there are multiple Dgeni export documents with the
+    // same document id. If there are multiple docs with the same id, we can assume that this doc
+    // is an alias export.
```

Comment:
Is there any way to more specifically target things marked as `@deprecated`? It's possible we may want to re-export something not as a deprecation, but as an alternate name (e.g. re-exporting something from cdk with "mat").

Reply:
I see. Unfortunately it's not possible to detect if the re-export was marked as deprecated.

```ts
/** @deprecated */
export {XX as PP} from './test';
```

Dgeni doesn't provide access to the export statement/symbol. In theory it would be possible to walk through imports file by file using the AST, but this more turns into it's own docs generation tool then.

I assume even Dgeni in general (with https://github.com/angular/dgeni-packages/issues/248) won't be able to differentiate between deprecated re-exports or normal re-exports.

Reply:
What if we formatted it like this?
```ts
export {XX as /** @deprecated */ PP} from './test';
```

Reply:
Just tried that, but that information is being lost as well.